src/main/java/eu/europeana/metis/sandbox/config/amqp/WorkflowListenerConfig.java

```diff
@@ -0,0 +1,84 @@
+package eu.europeana.metis.sandbox.config.amqp;
+
+import org.springframework.amqp.rabbit.config.SimpleRabbitListenerContainerFactory;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.support.converter.MessageConverter;
+import org.springframework.boot.autoconfigure.amqp.SimpleRabbitListenerContainerFactoryConfigurer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+class WorkflowListenerConfig {
+
+  private final MessageConverter messageConverter;
+
+  public WorkflowListenerConfig(
+      MessageConverter messageConverter) {
+    this.messageConverter = messageConverter;
+  }
+
+  @Bean
+  SimpleRabbitListenerContainerFactory createdFactory(
+      SimpleRabbitListenerContainerFactoryConfigurer configurer,
+      ConnectionFactory connectionFactory) {
+    return getSimpleRabbitListenerContainerFactory(configurer, connectionFactory);
+  }
+
+  @Bean
+  SimpleRabbitListenerContainerFactory externallyValidatedFactory(
+      SimpleRabbitListenerContainerFactoryConfigurer configurer,
+      ConnectionFactory connectionFactory) {
+    return getSimpleRabbitListenerContainerFactory(configurer, connectionFactory);
+  }
+
+  @Bean
+  SimpleRabbitListenerContainerFactory transformedFactory(
+      SimpleRabbitListenerContainerFactoryConfigurer configurer,
+      ConnectionFactory connectionFactory) {
+    return getSimpleRabbitListenerContainerFactory(configurer, connectionFactory);
+  }
+
+  @Bean
+  SimpleRabbitListenerContainerFactory internallyValidatedFactory(
+      SimpleRabbitListenerContainerFactoryConfigurer configurer,
+      ConnectionFactory connectionFactory) {
+    return getSimpleRabbitListenerContainerFactory(configurer, connectionFactory);
+  }
+
+  @Bean
+  SimpleRabbitListenerContainerFactory normalizedFactory(
+      SimpleRabbitListenerContainerFactoryConfigurer configurer,
+      ConnectionFactory connectionFactory) {
+    return getSimpleRabbitListenerContainerFactory(configurer, connectionFactory);
+  }
+
+  @Bean
+  SimpleRabbitListenerContainerFactory enrichedFactory(
+      SimpleRabbitListenerContainerFactoryConfigurer configurer,
+      ConnectionFactory connectionFactory) {
+    return getSimpleRabbitListenerContainerFactory(configurer, connectionFactory);
+  }
+
+  @Bean
+  SimpleRabbitListenerContainerFactory mediaProcessedFactory(
+      SimpleRabbitListenerContainerFactoryConfigurer configurer,
+      ConnectionFactory connectionFactory) {
+    return getSimpleRabbitListenerContainerFactory(configurer, connectionFactory);
+  }
+
+  @Bean
+  SimpleRabbitListenerContainerFactory indexedFactory(
+      SimpleRabbitListenerContainerFactoryConfigurer configurer,
+      ConnectionFactory connectionFactory) {
+    return getSimpleRabbitListenerContainerFactory(configurer, connectionFactory);
+  }
```

Comment:
This is 8 times exactly the same method... is there nothing we can do here?

Reply:
Each bean is for a different queue (and there is one queue per workflow step), code looks the same because right now all of this beans are using the same default config that I specified in the application.yml, like the individual threads they are using to consume from the queues. So while they all look the same they are defined for different queues, and the requirements of those queues might make the config for each of these beans to change (needing more consumers, or changing the retry policies). But I did encapsulate the repetitive code in the last method `getSimpleRabbitListenerContainerFactory`